### PR TITLE
add preproc steps

### LINKED ISSUE #6: Preprocessing:
Since we talked about it last time, I read the autoreject documentation again.
maybe the best workflow is:

1) hp filter, 
2) run autoreject (but not transform), 
3) then run ICA on "good channels" only

[ica](https://mne.tools/dev/generated/mne.preprocessing.ICA.html#mne.preprocessing.ICA) = [mne.preprocessing.ICA](https://mne.tools/dev/generated/mne.preprocessing.ICA.html#mne.preprocessing.ICA)(random_state=99)
[ica.fit](https://mne.tools/dev/generated/mne.preprocessing.ICA.html#mne.preprocessing.ICA.fit)([epochs](https://mne.tools/dev/generated/mne.Epochs.html#mne.Epochs)[~[reject_log.bad_epochs](https://numpy.org/devdocs/reference/generated/numpy.ndarray.html#numpy.ndarray)])

4) then detect classic components, and apply ica
5) run another autoreject and transform this time

sounds reasonable?

### PR DESCRIPTION
**Pull request by sappelhoff**
~~work in progress, as outlines in #6~~

closes #6 

- [x] bad segments
- [x] run ica
- [x] inspect ica
- [x] make epochs
- [x] run autoreject

The idea of this flow is to:

1. identify bad channels and bad segments automatically (using pyprep and MNE-Python methods)
2. run an ICA without the "bad" data, to get robust estimates of blink and saccade ICs
3. reject the blink and saccade ICs
4. Make long epochs (long enough for TFR analysis)
5. filter very conservatively (between 0.1 and 100Hz, with notch at line freq 50 Hz)
6. apply autoreject to more "elegantly" reject bad epochs (trials) and interpolate bad channels **locally** (i.e., per trial -- NOTE: the bad channels and segments from step 1 are ignored/discarded now: they only served the purpose to fit the ICA properly without too much noise)
7. interpolate all data to the common average reference

Additional considerations:
- the intraorbital channels are not considered for analysis (considered EOG channels, rather than EEG)
- instead of applying autoreject, one **could** simply interpolate the bad channels from step 1, and apply step 2 from the FASTER pipeline to reject epochs --> this potentially discards **more** epochs than the autoreject method ... not sure :shrug:

### CODE REVIEW

File: .github/workflows/run_analysis.yml [hunk 1]
```diff
@@ -107,7 +107,7 @@ jobs:
       # (only if not already restored from cache)
       if: steps.cache.outputs.cache-hit != 'true'
       run: |
-        datalad clone https://gin.g-node.org/sappelhoff/eeg_manypipes_arc $GITHUB_WORKSPACE
+        datalad clone https://gin.g-node.org/sappelhoff/eeg_manypipes_arc $GITHUB_WORKSPACE/eeg_manypipes_arc
```
Comment (NomisCiri) [inline .github/workflows/run_analysis.yml:107]: i steal this for my tests, too as long as we did not merge

**Comment by sappelhoff** [comment]
I think CIs are failing because of datalad stuff :-)

meanwhile I have pushed derivatives to our dataset repo ... so when we get that data here and "overwrite" it in the CI, datalad complains, because these files are protected. We first need to run `datalad unlock *` ... or use `datalad run <command> --output <file>`

**Review by NomisCiri — APPROVED** [review]

File: 05_make_epochs.py [hunk 2]
```diff
@@ -37,7 +37,7 @@
 fpath_ds = FPATH_DS
 overwrite = True
 
-t_min_max_epochs = (-1.0, 1.0)
+t_min_max_epochs = (-1.5, 2.5)
```
[the remark this answers is not in the document]
Comment (sappelhoff) [reply 05_make_epochs.py:40]: yeah, Casper said it was too short and then we decided to make them longer: https://chat.mpib-berlin.mpg.de/arc/pl/3uhbs537ziy5unjkn4c7y6betr

File: 03_run_ica.py [hunk 3]
```diff
@@ -1 +1,148 @@
-"""Highpass filter data and run ICA."""
```
[the remark this answers is not in the document]
Comment (sappelhoff) [reply 03_run_ica.py]: the plotted components should be in `*_report-ica.html` and the discarded components in `*_bad-components.json`

File: 03_run_ica.py [hunk 4]
```diff
@@ -1 +1,148 @@
...
+raw = raw.resample(sfreq=128)
+
+# %%
+# Initialize an ICA object, using picard with settings equal to extended infomax
+# Only estimate the first 20 components
+assert len(raw.info["projs"]) == 0
+ica = ICA(
+    n_components=20,
+    random_state=ica_rng,
+    method="picard",
```
[the remark this answers is not in the document]
Comment (sappelhoff) [reply 03_run_ica.py:120]: probably not used as widely as extended infomax, but it's generally faster and with the settings I select here, the "same problem" is solved as in extended infomax. Having inspected the results (I checked each by hand), I also trust this method

File: 04_inspect_ica.py [hunk 5]
```diff
@@ -1 +1,238 @@
...
+
+# %%
+# Plot components
+with plt.style.context("default"):
+    fig = ica.plot_components(inst=raw)
+
+# %%
+# Create VEOG epochs and plot evoked
+epochs_veog = create_eog_epochs(raw, ch_name="VEOG", picks="eeg")
+fig = epochs_veog.average().plot()
```
[the remark this answers is not in the document]
Comment (sappelhoff) [reply 04_inspect_ica.py:197]: yup, in the report :-)

**Pull request merged**